apache/superset
# feat(telemetry): Adding Scarf based telemetry to Superset

### SUMMARY

This PR does a handful of things:
* Adds the Scarf pixel to Superset (as approved by SIP) via a new `TelemetryPixel` component
* Adds the `ENABLE_TELEMETRY` feature flag 
* Adds basic tests for the above
* Bumps react types to support the `no-referrer-when-downgrade` 
* Bumps the npm version... not that I intended to, but it keeps coming along for the ride
* Fixes the shim for frontend tests, which had a weird incompatibility with certain node 18.x versions, throwing a `TypeError: Cannot redefine property: window` error

TODO:
* Get the pixel to load via Scarf gateway so that url params (version/sha/build) are reliably captured
* Feature flag documentation
* Turn off _more_ Scarf stuff with this flag?

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

N/A

### TESTING INSTRUCTIONS

RTL tests for the new component with FF on and off should pass. Run it, and have any PMC member check the Scarf data for proper capture. 

### ADDITIONAL INFORMATION


- [x] Has associated issue: https://github.com/apache/superset/issues/25639
